### PR TITLE
VEN-668 | Sending multiple offers

### PR DESCRIPTION
**Pull request by Nizar-Rahme**
### Closes :no_good_woman:
[**VEN-668:**](https://helsinkisolutionoffice.atlassian.net/browse/VEN-668) Add a dialog for sending offers

### Manual testing :construction_worker_man:
- Go to berth applications list view
- Select multiple applications
- "Lähteä tarjous" button at the top will be activated, click it to open a model 
- Set the due date and send the offers

Note: Selecting unhandled offers would display an error message to warn the admin.

## Screenshots :camera_flash:
image
image

### CODE REVIEW

File: src/features/applicationList/ApplicationList.tsx [hunk 1]
```diff
@@ -142,6 +154,32 @@ const ApplicationList = ({
...
+
+          let toastId;
+          if (offersWithoutPlacesCount > 0)
+            toastId = hdsToast({
+              type: 'error',
+              toastId: 'multiApplicationsError',
+              labelText: 'applicationList.errors.unhandledApplications.label',
+              text: 'applicationList.errors.unhandledApplications.description',
+              translated: true,
+            });
```
Comment (TatuArvela) [inline src/features/applicationList/ApplicationList.tsx:169]: Worked nicely, but I couldn't read it in time 😄 Maybe disable auto-dismiss for this one, with the prop `autoDismiss: false`

File: src/locales/fi.json [hunk 2]
```diff
@@ -258,6 +258,17 @@
       "phone": "Puhelinnumero",
       "email": "Sähköpostiosoite",
       "comment": "Huomiot"
+    },
+    "sendMultiOffers": {
+      "title": "Lähetä Tarjous",
```
Comment (TatuArvela) [inline src/locales/fi.json:263]: Minor typo, should be "Lähetä tarjous"

File: src/locales/fi.json [hunk 3]
```diff
@@ -308,6 +319,12 @@
   "applicationList": {
     "title": "Venepaikkahakemukset",
     "winterStorageTitle": "Talvisäilytyspaikkahakemukset",
+    "tools": {
+      "sendOffer": "Lähteä tarjous",
```
Comment (TatuArvela) [inline src/locales/fi.json:323]: Typo, should be "Lähetä tarjous"

**Review by TatuArvela — CHANGES_REQUESTED** [review]
A couple of things to correct, but otherwise it's looking good. Very nicely done! 👍

**Comment by TatuArvela** [comment]
I rebased this, but I definitely need another set of eyes to look at this. @ahie or @santtul ?

File: src/features/applicationList/ApplicationListContainer.tsx [hunk 4]
```diff
@@ -41,6 +48,31 @@ const ApplicationListContainer = () => {
...
+          },
+        },
+      });
+      hdsToast({
+        type: 'success',
+        labelText: 'applicationList.notifications.offersSent.label',
+        text: 'applicationList.notifications.offersSent.description',
+        translated: true,
+      });
+    } catch (error) {}
```
Comment (ahie) [inline src/features/applicationList/ApplicationListContainer.tsx:73]: Should an error toast be shown here?

File: src/features/unmarkedWsNoticeList/UnmarkedWsNoticeList.tsx [hunk 5]
```diff
@@ -99,6 +110,32 @@ const UnmarkedWsNoticeList = ({
             <UnmarkedWsNoticeDetails {...row.original} />
           </Grid>
         )}
+        renderTableToolsTop={({ selectedRows }, { resetSelectedRows }) => {
+          const offers = getDraftedOffers(selectedRows);
```
Comment (ahie) [inline src/features/unmarkedWsNoticeList/UnmarkedWsNoticeList.tsx:114]: Should this logic be in `ApplicationListTools` instead? Now it's still possible to send the BE request even though `offers` contains ones without places?

File: src/features/applicationListTools/ApplicationListTools.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+import Text from '../../common/text/Text';
+import SendMultiOffersForm from '../sendMultiOffersForm/SendMultiOffersForm';
+import styles from './applicationListTools.module.scss';
+
+export interface ApplicationListToolsProps {
+  offersCount: number;
+  selectedApplicationsCount: number;
+  isSubmitting: boolean;
+  clearSelectedRows(): void;
+  handleApproveOrders(): Promise<void>;
```
Comment (ahie) [inline src/features/applicationListTools/ApplicationListTools.tsx:15]: Optional: Not sure about the `Promise<void>` return type here, as we usually just pass synchronous functions to components. Could refactor?

File: src/features/applicationList/ApplicationList.tsx [hunk 7]
```diff
@@ -164,6 +176,32 @@ const ApplicationList = ({
             />
           );
         }}
+        renderTableToolsTop={({ selectedRows }, { resetSelectedRows }) => {
+          const offers = getDraftedOffers(selectedRows);
```
Comment (ahie) [inline src/features/applicationList/ApplicationList.tsx:180]: Same comment here as in unmarked ws notice list: the additional logic could be moved to the `ApplicationListTools` component?

**Review by ahie — COMMENTED** [review]
Left a couple comments, otherwise LGTM.

File: src/features/applicationList/ApplicationListContainer.tsx (hunk 4 above)
Comment (ahie) [reply src/features/applicationList/ApplicationListContainer.tsx:73]: Typically these handlers are synchronous and the success toast is triggered in a `then` block, while the error is handled by the generic error handler.

**Review by TatuArvela — APPROVED** [review]

**Comment by TatuArvela** [comment]
Rebased to `develop`. Had to rewrite the commits a bit to make this cleaner. Some changes were on the wrong commits when compared to commit titles, and did not pass tests or type checks.

**Review by ahie — APPROVED** [review]
LGTM 👍

**Pull request merged**